code/datums/minimap.dm

```diff
@@ -0,0 +1,120 @@
+/datum/minimap
+	///The map render to be displayed.
+	var/icon/map_render
+
+	///The z-level that the map is to be rendered from.
+	var/z_level = null
+	///The maximum x coordinate to be rendered, in world coordinates.
+	var/x_max = null
+	///The minimum x coordinate to be rendered, in world coordinates.
+	var/x_min = null
+	///The maximum y coordinate to be rendered, in world coordinates.
+	var/y_max = null
+	///The minimum y coordinate to be rendered, in world coordinates.
+	var/y_min = null
+
+	New()
+		. = ..()
+		if (!map_render)
+			src.render_map()
+
+	///Renders the map within the boundaries defined by x_max, x_min, y_max, and y_min.
+	proc/render_map()
+		if (!x_max || !x_min || !y_max || !y_min || !z_level)
+			return
+		src.map_render = icon('icons/obj/station_map.dmi', "blank")
+		for (var/turf/T in block(locate(src.x_min, src.y_min, src.z_level), locate(src.x_max, src.y_max, src.z_level)))
+			if (!src.valid_turf(T))
+				continue
+			map_render.DrawBox(turf_color(T), T.x, T.y)
+
+	///Checks whether a turf should be rendered on the map through the render_on_map variable on /turf.
+	proc/valid_turf(var/turf/T)
```

Comment:
This is missing the horrible yet necessary snowflake checks for map specific off station areas that we don't want the AI map to render. If you want them to render on the generic Z level map then make a subtype specifically for the AI one, as it really messes with the zooming on certain maps.

Reply:
Following feedback from Tarmunora, I opted to remove them, as while they may not be necessary for the AI map, other maps should have them.